Rithika taking over for Kanishk: Implement Event Participation Analytics landing page with functional dropdowns and CSS modules

kanishkagarwal6101 opened the pull request:
# Description
Implemented the Event Participation Analytics landing page at `/communityportal/reports/participation` with comprehensive analytics dashboard, functional dropdowns, and CSS modules conversion. This addresses the urgent issue where the event registration page was only showing month/year view and the registration form/page was not accessible.

Fixes # (Event Participation Analytics landing page accessibility and functionality)

## Related PRS (if any):
This frontend PR implements the complete Event Participation Analytics landing page as specified in the requirements.

## Main changes explained:
- **Fixed routing** in `src/routes.jsx` to make `/communityportal/reports/participation` accessible
- **Converted to CSS modules** for all participation components:
  - `MyCases.jsx` → uses `MyCases.module.css`
  - `DropOffTracking.jsx` → uses `Participation.module.css`
  - `NoShowInsights.jsx` → uses `Participation.module.css`
- **Created comprehensive analytics dashboard** with new components:
  - `EventParticipationHeader.jsx` - Header with navigation and event metrics summary
  - `EngagementSummaryCards.jsx` - Key performance indicator cards
  - `EventTypePieChart.jsx` - Pie chart for event type popularity visualization
  - `EngagementBarChart.jsx` - Bar chart for monthly engagement trends
  - `AnalyticsNavigation.jsx` - Navigation links to detailed analytics sub-pages
- **Added four new analytics sub-pages** with functional dropdowns:
  - `VirtualVsInPerson.jsx` - Compare virtual vs in-person attendance patterns
  - `EventValue.jsx` - Analyze estimated event value and ROI calculations
  - `ParticipationTrends.jsx` - Track participation growth over time
  - `EventPerformance.jsx` - Detailed individual event performance metrics
- **Created CSS modules** with dark mode support:
  - `Participation.module.css` - Main styles for all participation components
  - `MyCases.module.css` - Specific styles for MyCases component
- **Implemented functional dropdowns** with dynamic data filtering for all analytics pages

## How to test:
1. Check into current branch `kanishk_design_the_landing_page_analytics_eventPopularity`
2. Run `npm install` to install dependencies
3. Run `npm start` to start the app locally
4. Clear site data/cache
5. Log in as any user
6. Navigate to `/communityportal/reports/participation`
7. Verify the landing page loads with all sections:
   - Header with navigation and metrics summary
   - Engagement summary cards
   - Pie chart and bar chart visualizations
   - Analytics navigation section
8. Test navigation links to sub-pages:
   - Virtual vs. In-Person Attendance
   - Estimated Event Value
   - Participation Trends
   - Event Performance Metrics
9. Test dropdown functionality on each sub-page to ensure data changes dynamically
10. Test dark mode toggle to verify all components support theme switching

## Screenshots or videos of changes:
<img width="1904" height="995" alt="Screenshot 2025-10-03 at 7 54 44 PM" src="https://github.com/user-attachments/assets/2242ca5e-62b5-4d4a-a6d9-2411c19ad0b9" />
<img width="1901" height="888" alt="Screenshot 2025-10-03 at 7 58 45 PM" src="https://github.com/user-attachments/assets/a401be18-3ef6-4fd7-862d-5847733eb502" />


## Note:
- All components use CSS modules for better maintainability and to prevent class name collisions
- Mock data is used for demonstration purposes and should be replaced with real API data
- Dropdowns are fully functional with dynamic data filtering
- All navigation links are working and properly routed
- The page serves as a central hub for accessing event participation data and analytics

kanishkagarwal6101 committed:
Kanishk: Implement Event Participation Analytics landing page with functional dropdowns and CSS modules

kanishkagarwal6101 committed:
Kanishk: Implement Event Participation Analytics landing page with functional dropdowns and CSS modules

sumedhkumar96 reviewed (changes requested):
Followed the instructions provided by the developer and verified all functionalities as stated:

#### 1. Landing Page
✅ Header with navigation and metrics summary  
<img width="1920" height="980" alt="header and summary" src="https://github.com/user-attachments/assets/7d8c7c75-2779-4519-ae16-a0ff62d4b379" />

✅ Engagement summary cards  
<img width="1920" height="980" alt="header and summary" src="https://github.com/user-attachments/assets/5427ba04-650f-4fd7-be37-46a619564dab" />

✅ Pie chart and bar chart visualizations  
<img width="1920" height="975" alt="piechart" src="https://github.com/user-attachments/assets/e7dffa06-b2a4-4117-9bfe-15b47d0d16b8" />

✅ Analytics navigation section  
<img width="1920" height="977" alt="analytics navigation section" src="https://github.com/user-attachments/assets/0f50aa61-f5f8-460f-93ee-09d352c1cbe2" />


#### 2. Navigation Links to Sub-Pages
✅ Virtual vs. In-Person Attendance  
<img width="1920" height="968" alt="virtual" src="https://github.com/user-attachments/assets/5c85bd4c-e30a-43e6-bf5f-729aa4e3c49d" />

✅ Estimated Event Value  
<img width="1920" height="982" alt="estimated" src="https://github.com/user-attachments/assets/33ecd008-286a-4f43-94b1-d1a9fdf3783b" />

✅ Participation Trends  
<img width="1920" height="980" alt="participation" src="https://github.com/user-attachments/assets/87e79909-1c6d-4337-9beb-4991adb7098a" />

✅ Event Performance Metrics  
<img width="1920" height="978" alt="event" src="https://github.com/user-attachments/assets/f29dc6f8-6443-4a03-9c30-71d7b1850b41" />


#### 3. Dropdown Functionality
✅ Verified that dropdowns on each sub-page dynamically update the displayed data  

#### 4. Dark Mode Toggle
**❌ Tested dark mode toggle, some fonts **do not fully support** theme switching**  
<img width="1920" height="978" alt="Screenshot 2025-10-08 162618" src="https://github.com/user-attachments/assets/b7cd03c3-bc8f-4e61-82fa-a47d05a0e008" />
<img width="1920" height="982" alt="Screenshot 2025-10-08 171845" src="https://github.com/user-attachments/assets/01931e3d-7ce9-4aea-9d84-98efc8ae2fcb" />
<img width="1920" height="977" alt="Screenshot 2025-10-08 172022" src="https://github.com/user-attachments/assets/dbd7c551-cd8a-44d3-bf6f-2a758edeaf9b" />

Aswin20010 reviewed (changes requested):
Tested the landing page at /communityportal/reports/participation and verified that all sections — header, summary cards, charts, and navigation to sub-pages — load and function correctly. Dropdowns work as expected, and data updates dynamically across views.
However, in dark mode, the font contrast is low, making text and labels less visible in several areas. Functionality is solid, but readability needs improvement. Please enhance text contrast for better visibility before merging.

<img width="1512" height="982" alt="Screenshot 2025-10-15 at 1 10 54 PM" src="https://github.com/user-attachments/assets/737ff2ae-e72b-46d3-b2f9-af82773f5b22" />
<img width="1512" height="982" alt="Screenshot 2025-10-15 at 1 11 05 PM" src="https://github.com/user-attachments/assets/7b785caa-9839-4151-addd-cf12e2605d91" />
<img width="1512" height="982" alt="Screenshot 2025-10-15 at 1 11 19 PM" src="https://github.com/user-attachments/assets/74a60000-6c73-484a-b34b-da0f2b4a2086" />

Aditya-gam reviewed (changes requested):
- Tested the changes mentioned in this PR. All the links and pages work correctly and are visually readable in light mode.
- In dark mode, some of the titles and labels on the charts are dark-colored, making them difficult to see.
- The three-button links have no backlinks to the landing page.

https://github.com/user-attachments/assets/d980b30c-2abd-4a02-a12a-57899b2dc590

kanishkagarwal6101 committed:
Fix dark mode contrast and layout issues for Event Participation Analytics

- Fixed dark mode text contrast across all components (cards, charts, tables)
- Improved hover states for dark mode (no more white backgrounds)
- Fixed dark mode horizontal stretching on sub-pages (100% viewport width)
- Left-aligned main heading on landing page for better alignment
- Enhanced readability with proper color contrast in dark mode
- Fixed backlink button visibility in light mode
- Ensured all content remains centered while dark background stretches full width

kanishkagarwal6101 committed:
Fix dark mode contrast and layout issues for Event Participation Analytics

- Fixed dark mode text contrast across all components (cards, charts, tables)
- Improved hover states for dark mode (no more white backgrounds)
- Fixed dark mode horizontal stretching on sub-pages (100% viewport width)
- Left-aligned main heading on landing page for better alignment
- Enhanced readability with proper color contrast in dark mode
- Fixed backlink button visibility in light mode
- Ensured all content remains centered while dark background stretches full width

kanishkagarwal6101 committed:
Merge conflict fixes

kanishkagarwal6101 committed:
Remove print-only heading and footer from participation landing page

kanishkagarwal6101 committed:
Remove print-only heading and footer from participation landing page

Aditya-gam reviewed (dismissed):
- Tested the changes mentioned in this PR. All links and pages work correctly and are visually readable in light mode.

<img width="1512" height="910" alt="Image" src="https://github.com/user-attachments/assets/1d8380ea-c053-4f90-a9cd-1efe5cec1e88" />

- In dark mode, all titles and labels on the charts are now clearly readable with proper contrast.
- The comprehensive dark mode CSS fixes ensure that chart legends, axis labels, bar labels, and all text elements have sufficient contrast against the dark background.

<img width="1512" height="910" alt="Image" src="https://github.com/user-attachments/assets/e5ac11fc-5c26-4c6f-b66a-5521de88242d" />

- All analytics sub-pages (Virtual vs. In-Person Attendance, Estimated Event Value, Participation Trends, and Event Performance Metrics) now include backlinks to the participation landing page, improving navigation flow.

<img width="523" height="540" alt="Image" src="https://github.com/user-attachments/assets/ea1f9b03-1457-42e9-a1d6-85be3f95d808" />

**Testing:**

Verified functionality across all pages:
- Landing page loads with all sections (header, engagement cards, charts, navigation)
- All four analytics sub-pages are accessible and functional
- Dropdown filters work correctly with dynamic data updates
- Dark mode toggle works properly throughout all components
- Navigation links function as expected

Test video demonstrating the functionality:

https://github.com/user-attachments/assets/7d7d2e0d-e740-464b-bb40-5d11d1b87d99

Note:
- You should remove yarn.lock from the commit so there are no merge conflicts with development.

Anusha-Gali reviewed (changes requested):
Hi Kanishk,

I have reviewed your PR locally and most of the functionalities do work as per requirement but there are a few misses in the dark mode (point 1,2,3,4,5) and one in light mode(point 4):
**In Estimated Event Value Analysis - The word Value in Monthly Value Trends overlaps the X-axis and the title and there is no data label for Y axis**
<img width="1271" height="754" alt="Screenshot 2025-12-03 at 8 51 26 PM" src="https://github.com/user-attachments/assets/95a02d1f-4e67-4495-95ff-76bfbb64ec39" />
**In Participation Trends Analysis - The word Attendance in Monthly Participation Trends overlaps the title and there is no data label for Y axis**
<img width="1271" height="754" alt="Screenshot 2025-12-03 at 8 51 48 PM" src="https://github.com/user-attachments/assets/6347daeb-e6db-402a-9d16-bb62f19e1e11" />
**On the dashboard - The word Attendance in Monthly Engagement Trends overlaps the title**
<img width="1271" height="754" alt="Screenshot 2025-12-03 at 8 52 07 PM" src="https://github.com/user-attachments/assets/288a31ed-739b-4cd6-8b2e-6770948f9a24" />
**In Event Performance Metrics - When the Upcoming Events dropdown is selected in the Filter Events - the dates do not co-relate to the current system date**
<img width="1271" height="754" alt="Screenshot 2025-12-03 at 8 52 52 PM" src="https://github.com/user-attachments/assets/7a1dddf5-9a06-4cac-b074-d9c42d337179" />
**On the dashboard - In Upcoming Events - When selected filters that also show the words "More" and "Show Less" - their text color is black and does not correlate to dark mode**
<img width="1271" height="338" alt="Screenshot 2025-12-03 at 8 53 01 PM" src="https://github.com/user-attachments/assets/e08f3f1e-90fc-41e2-91ce-8c5a3dbef042" />
<img width="1271" height="612" alt="Screenshot 2025-12-03 at 8 53 50 PM" src="https://github.com/user-attachments/assets/876c870d-c149-4d7b-a6f3-deb837076901" />

kanishkagarwal6101 committed:
Fix dark mode text colors for drop-off rate and no-show rate labels

- Add dark mode styling for 'Drop-off rate' and 'No-show rate' text in DropOffTracking component
- Fix duplicate CSS selectors to resolve stylelint errors
- Fix CSS syntax error (extra closing brace)
- Ensure text is visible in dark mode with proper color contrast

kanishkagarwal6101 committed:
Fix dark mode text colors for drop-off rate and no-show rate labels

- Add dark mode styling for 'Drop-off rate' and 'No-show rate' text in DropOffTracking component
- Fix duplicate CSS selectors to resolve stylelint errors
- Fix CSS syntax error (extra closing brace)
- Ensure text is visible in dark mode with proper color contrast

kanishkagarwal6101 committed:
Merge development branch - resolve yarn.lock conflict

- Resolved yarn.lock merge conflict by accepting development version and regenerating
- Merged latest changes from development branch

debadyuti23 reviewed (changes requested):
Hi @kanishkagarwal6101 after landing the webpage, these are my observations in the video shown below.

https://github.com/user-attachments/assets/7871d1ec-3860-4d93-a592-16ac70df4ba1

In short:
1. every functionality is working
2. in dark mode some boxes dont render with appropriate color. Here's a screenshot of the broken dark mode
<img width="1262" height="617" alt="PR_4227_dark_mode_bug" src="https://github.com/user-attachments/assets/73d6c0de-b6c5-4a7c-b447-687d3ad7eeb2" />
Therefore would request to resolve this bug since other reviewers are also having issues with dark mode

rohanrastogi311 reviewed (changes requested):
Hi Kanishk, great work with the page. All functionality except dark mode is working great. My video below of the run through showcases this as well as highlights the issues with the dark mode toggle. Some of the dark mode toggle issues are: The boxes for no-show rate insights and Drop off and now show rate tracking.

Video of my run-through highlighting the issues: https://www.dropbox.com/scl/fi/hexzbsxh9am08co8u3i74/PR-4162-Recording.mov?rlkey=72bphz7hje7vlhghuyzhrmhlw&st=6b8i756k&dl=0

kanishkagarwal6101 committed:
fix: improve dark mode readability for event badges, view switcher buttons, and filter dropdowns

Anusha-Gali reviewed (approved):
Hi Kanishk,

I have re-reviewed the PR locally and all the errors i have pointed appear to be fixed except i could not test The Show More or Show Less probably due to the counts of event being less.
<img width="1258" height="739" alt="Screenshot 2025-12-16 at 10 10 50 PM" src="https://github.com/user-attachments/assets/aea1d4fc-ac66-4cf0-8d9d-e60872274a87" />
<img width="1258" height="739" alt="Screenshot 2025-12-16 at 10 11 17 PM" src="https://github.com/user-attachments/assets/ddd5d860-9827-43c0-a284-db53eb0de242" />
<img width="1258" height="739" alt="Screenshot 2025-12-16 at 10 11 28 PM" src="https://github.com/user-attachments/assets/ea342cc0-6979-4bba-a94f-6d9790772ceb" />
<img width="1258" height="739" alt="Screenshot 2025-12-16 at 10 12 19 PM" src="https://github.com/user-attachments/assets/b8a2198a-29ff-4d50-83f5-ed5e1c13ac9f" />
<img width="1272" height="739" alt="Screenshot 2025-12-16 at 10 13 20 PM" src="https://github.com/user-attachments/assets/743cdbe2-7004-45f8-a98f-298b99b09968" />

rohanrastogi311 reviewed (approved):
Hi Kanishk, 

Well done on the fix. Dark mode issues are rectified. 

<img width="1713" height="1031" alt="PR 4162 Screenshot" src="https://github.com/user-attachments/assets/0eb51ef4-4142-4539-9094-51acb8aa9ab3" />
<img width="1728" height="1032" alt="PR 4162 Screenshot 2" src="https://github.com/user-attachments/assets/ac20f5a8-2e4c-4b65-b19e-3fb84c781d88" />

Vikas-8055 reviewed (approved):
Hi Kanishk,
I have thoroughly tested this PR locally and all functionalities work as expected. The landing page loads correctly with all sections (header, engagement cards, charts, and navigation links). All sub-pages (Virtual vs. In-Person, Event Value, Participation Trends, Event Performance) navigate properly and display data correctly. Dropdown filters work smoothly and update data dynamically. Dark mode implementation is excellent - all text, charts, and labels have proper contrast and are clearly readable with no overlap issues. Responsive design adapts well to different screen sizes. Overall, the analytics dashboard is clean, intuitive, and fully functional.

<img width="1511" height="858" alt="Screenshot 2026-01-08 at 2 00 53 PM" src="https://github.com/user-attachments/assets/dc318326-9c01-45b0-b7a7-0424d67e33d5" />
<img width="1512" height="848" alt="Screenshot 2026-01-08 at 2 01 00 PM" src="https://github.com/user-attachments/assets/79249b2d-0b58-4c9f-b9bf-6680631f7726" />
<img width="1512" height="815" alt="Screenshot 2026-01-08 at 2 01 11 PM" src="https://github.com/user-attachments/assets/6d79ce6d-bd4f-4c69-8781-989c7f620e37" />
<img width="1510" height="867" alt="Screenshot 2026-01-08 at 2 01 23 PM" src="https://github.com/user-attachments/assets/f6144f5c-9587-4c9e-9469-6e1af8cce3b1" />
<img width="1509" height="859" alt="Screenshot 2026-01-08 at 2 01 54 PM" src="https://github.com/user-attachments/assets/b71de889-adfb-467e-957b-a98a04d06fba" />

Shravan-neelamsetty reviewed (approved):
Hi Kanishk, I tested this PR locally by navigating to /communityportal/reports/participation and verified all the functionalities. The landing page loads correctly with all sections including the header, engagement summary cards, pie chart, bar chart visualizations, and analytics navigation. All four sub-pages (Virtual vs. In-Person Attendance, Estimated Event Value, Participation Trends, and Event Performance Metrics) are accessible and working properly. The dropdown filters work smoothly and update data dynamically. Dark mode implementation looks good with proper text contrast across all pages and charts. Backlinks to the landing page are working as expected.

<img width="1470" height="829" alt="Screenshot 2026-01-08 at 5 37 11 PM" src="https://github.com/user-attachments/assets/9db92d7f-cb34-486a-9791-d06b1d7c35aa" />
<img width="1470" height="828" alt="Screenshot 2026-01-08 at 5 37 20 PM" src="https://github.com/user-attachments/assets/1c457477-d69c-4773-8a43-015d5a6d5bb2" />
<img width="1470" height="830" alt="Screenshot 2026-01-08 at 5 37 47 PM" src="https://github.com/user-attachments/assets/61c611a7-1ab9-4d00-9793-5b1e052d064f" />
<img width="1470" height="832" alt="Screenshot 2026-01-08 at 5 37 56 PM" src="https://github.com/user-attachments/assets/b9085db1-ffa4-48f8-804a-d87baab9831f" />
<img width="1470" height="832" alt="Screenshot 2026-01-08 at 5 38 07 PM" src="https://github.com/user-attachments/assets/144516c7-dc85-4a29-939d-7d4949541946" />

Ganesh112001 reviewed (approved):
Hi Kanishk, I tested this PR locally following the provided steps and verified that all functionality works perfectly as intended. After checking out the kanishk_design_the_landing_page_analytics_eventPopularity branch, installing dependencies, and starting both servers, I navigated to /communityportal/reports/participation and confirmed the landing page is now fully accessible with all components rendering correctly. The comprehensive analytics dashboard displays beautifully with the header showing event metrics summary, engagement summary KPI cards, pie chart for event type popularity, bar chart for monthly engagement trends, and navigation links to all four sub-pages. I thoroughly tested each sub-page (Virtual vs. In-Person Attendance, Estimated Event Value, Participation Trends, and Event Performance Metrics) and verified that all functional dropdowns work smoothly with data updating dynamically based on selections. The CSS modules conversion is clean with consistent styling across all components, and dark mode support works flawlessly across every page with proper contrast and theme adaptation. No console errors were observed during testing, and the routing fix successfully makes the page accessible via direct URL. This is a comprehensive and polished implementation that successfully resolves the urgent accessibility issue while delivering a complete, professional analytics solution. Everything functions perfectly and is ready for merge.
<img width="1470" height="880" alt="Screenshot 2026-01-10 at 7 08 59 PM" src="https://github.com/user-attachments/assets/983ce696-3e48-49fd-9240-99bc83d5c686" />
<img width="1470" height="881" alt="Screenshot 2026-01-10 at 7 10 42 PM" src="https://github.com/user-attachments/assets/8e30dd38-febd-4adb-af5e-a1e3ae454c10" />

<img width="1468" height="876" alt="Screenshot 2026-01-10 at 7 11 06 PM" src="https://github.com/user-attachments/assets/a6582b12-c72a-4256-b471-14ff197a7415" />

abhinav-TB commented:
Hi Kanishk, I am reviewed you PR and everything works well as expected.

<img width="2560" height="1440" alt="image" src="https://github.com/user-attachments/assets/1858e6d2-a37a-4413-b087-92788993df79" />
<img width="2413" height="1273" alt="image" src="https://github.com/user-attachments/assets/e556656e-cae6-4332-b85c-ab5af14d7387" />

saitejakaasoju reviewed (approved):
Hi Kanishk,
I tested the landing page at /communityportal/reports/participation and confirmed that all major sections—including the header, summary cards, charts, and navigation to analytics sub-pages—load and function as expected. The dropdowns behave correctly, and the data updates dynamically across different views. Overall, I’ve reviewed your PR and everything works well as intended.
<img width="1440" height="858" alt="PR-4162" src="https://github.com/user-attachments/assets/33903ee3-65d4-4dfe-8fa5-26917c25bd8d" />
<img width="1440" height="854" alt="PR-4162-1" src="https://github.com/user-attachments/assets/8bf77287-31a1-4dbb-b7d6-517e73ce7829" />
<img width="1436" height="853" alt="PR-4162-2" src="https://github.com/user-attachments/assets/fb30bfdd-19bf-46b2-b935-5e0be43a8c06" />
<img width="1440" height="854" alt="PR-4162-3" src="https://github.com/user-attachments/assets/049e9995-7b77-4180-9a9c-a2b8879ac579" />

roshini1212 reviewed (approved):
Tested the landing page at `/communityportal/reports/participation`—all sections load correctly: header with metrics, engagement cards, pie/bar charts, and analytics navigation. Navigation to sub-pages (Virtual vs. In-Person, Event Value, Trends, Metrics) works, dropdowns update data dynamically, and all components support light/dark mode switching perfectly.

<img width="1512" height="865" alt="Image" src="https://github.com/user-attachments/assets/3e16d4c6-bbff-4df7-adc7-de6f9f501512" />

<img width="1512" height="856" alt="Image" src="https://github.com/user-attachments/assets/c95641ec-1adf-44fd-8411-6402883621e4" />

maithili20 commented:
Hi @kanishkagarwal6101 
Everything seems to work fine in both light and dark mode. The links to detailed analytics work fine and also the drop downs on those respective pages. The page is responsive as well. 
<img width="1807" height="864" alt="Screenshot 2026-02-09 at 6 41 39 PM" src="https://github.com/user-attachments/assets/b14fc976-dccf-49e2-a086-81a095917479" />
<img width="909" height="746" alt="Screenshot 2026-02-09 at 6 41 47 PM" src="https://github.com/user-attachments/assets/f985d3fd-6424-4008-adba-8faddaab3d26" />
<img width="1003" height="862" alt="Screenshot 2026-02-09 at 6 42 43 PM" src="https://github.com/user-attachments/assets/5955e18e-11ee-4ec5-a2fc-97a39fdfec41" />
<img width="971" height="975" alt="Screenshot 2026-02-09 at 6 42 59 PM" src="https://github.com/user-attachments/assets/1f4ae556-d3e2-4294-97db-dee5991dc680" />
<img width="1899" height="807" alt="Screenshot 2026-02-09 at 6 43 27 PM" src="https://github.com/user-attachments/assets/a19aecb3-1a28-4f7b-bf67-e060c3222be5" />

Vishnu10-hub commented:
- Ran both backend and frontend locally 
- Logged in and opened the participation analytics page
- Landing page loaded correctly
- All sections and navigation tabs are visible
- Sub-pages open properly
- Dropdown filters are working
- Charts update based on selected options

No UI or functional issues observed
<img width="1433" height="810" alt="Screenshot 2026-02-27 at 17 30 25" src="https://github.com/user-attachments/assets/fff00deb-c826-46f8-9b40-bee30c18a467" />
<img width="1433" height="810" alt="Screenshot 2026-02-27 at 17 30 32" src="https://github.com/user-attachments/assets/4cfaad26-257c-479f-b0bb-4041ef172c29" />
<img width="1433" height="810" alt="Screenshot 2026-02-27 at 17 30 46" src="https://github.com/user-attachments/assets/ee4d7251-60d3-4e8d-a5d1-237653be9655" />

naznin07 reviewed (approved):
Virtualized scrolling if events grow large. 
<img width="1807" height="517" alt="PR #4162" src="https://github.com/user-attachments/assets/4a10b93f-fe77-41a4-b1d9-2623993c9d17" />

ManojPuttaswamy reviewed (approved):
Landing page loads fine with all sections, navigation links are working as expected, UI looks fine in dark mode.

**Suggestion:** Using a plain <a> tag causes a full page reload instead of a client-side navigation.  Use React Router, <Link to={item.link}> and also using array index as key is an anti-pattern. Since navigationItems has stable titles, key={item.title} would be more appropriate.


<img width="1816" height="955" alt="Screenshot 2026-03-16 at 9 08 17 PM" src="https://github.com/user-attachments/assets/a481b6be-8fb1-495c-8b1d-c64aafe15343" />
<img width="1816" height="955" alt="Screenshot 2026-03-16 at 9 09 32 PM" src="https://github.com/user-attachments/assets/7dd5cfd2-fd01-4c37-8968-ca73328567db" />
<img width="1816" height="955" alt="Screenshot 2026-03-16 at 9 10 01 PM" src="https://github.com/user-attachments/assets/01710aeb-0b84-4296-8d2a-d7ef7a44c42d" />

SharadhaKasiviswanathan commented:
Tested the initial/default flow and everything else looks fine from my side. The only issue I found is in the Virtual vs In-Person Attendance page:
In the default 6-month view, the Growth Rate row shows '+18%' for Virtual and '+8%' for In-Person, but the winner is rendered as In-Person. This looks like the percentages are being compared as strings instead of numbers.
I think the growth values could be parsed to numbers before comparing them so the winner is calculated correctly.
<img width="1211" height="883" alt="Screenshot 2026-03-29 at 6 35 50 AM" src="https://github.com/user-attachments/assets/b54ed8d7-2e05-4506-b352-2822c2d9ecb1" />

rajanidi1999 reviewed (approved):
Hi,
I have tested your PR and its running successfully
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/c3fdb35a-4299-4032-8431-6b75ccdf37ba" />
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/06314e69-932b-4d2b-9ddf-76955a36398d" />

rithika-paii reviewed (approved):
Hi Vikas,

I reviewed this PR #4162 and tested the changes locally in both light and dark modes, as well as across different screen sizes. The functionality works as expected, and I did not encounter any issues during testing.

Approved from my end.

https://github.com/user-attachments/assets/3dec3c3e-28b3-45b0-9679-0222fc870fff

kunchalasireesha reviewed (changes requested):
Hi Vikas, This PR has enough reviews to merge but it looks like it currently has merge conflicts with the base branch. Could you please rebase or merge the latest changes from development into your branch and resolve the conflicts? Once the build passes, I’ll finish the review.

saurabhdipte reviewed (approved):
Hi Kanishk,

I tested and reviewed your changes locally, everything looks good. The sublinks are working fine, and each dropdown menu is functioning correctly. The pie chart and bar graphs are displaying proper headings as expected, and dark mode is working fine as well.

<img width="1859" height="990" alt="Image" src="https://github.com/user-attachments/assets/90def85a-2870-472f-95e3-f701fbc4adc4" />

<img width="1918" height="987" alt="Image" src="https://github.com/user-attachments/assets/be184571-afc2-42db-b971-92cc50329217" />

DeepighaJ reviewed (changes requested):
Checked out to the current branch and ran PR locally.
When navigating to /communityportal/reports/participation its asking to login.
Entered the credentials but no action happening on clicking submit.
No error msg or wrong credential or successful login.

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/eea84521-f6d3-46dd-a70b-ee622969b797" />

rithika-paii committed:
fix: remove duplicate imports and stale unused imports in routes.jsx introduced during rebase

rithika-paii committed:
chore: regenerate yarn.lock

rithika-paii committed:
fix: prettier formatting in EventParticipation.jsx

rithika-paii committed:
fix: resolve SonarQube code smells - unused imports, nested ternaries, array index keys, contrast issues, duplicate selectors

rithika-paii committed:
fix: prettier formatting in VirtualVsInPerson.jsx

rithika-paii committed:
fix: address remaining SonarQube contrast and array-index-key issues

rithika-paii committed:
fix: remove duplicate brace causing syntax error

rithika-paii committed:
fix: use solid colors for navLink to resolve contrast issues

kzou55 reviewed (changes requested):
Hi Rithika, 

I ran and tested the current branch locally.

## Verified
- Verify the landing page loads with all sections:
  - Header with navigation and metrics summary
  - Engagement summary cards
  - Pie chart and bar chart visualizations
  - Analytics navigation section
- The navigation links to sub-pages(Virtual vs. In-Person Attendance, Estimated Event Value, Participation Trends, Event Performance Metrics) are functional
- The dropdown functionality on each sub-page works
- Works in Dark Mode

    <img width="1905" height="1031" alt="Image" src="https://github.com/user-attachments/assets/d415b78f-701f-4f0e-80ca-27393f8c7beb" />

    <img width="1835" height="983" alt="Image" src="https://github.com/user-attachments/assets/3b4951c5-4a53-4605-a4e7-b434f124c3ba" />

    <img width="1916" height="1033" alt="Image" src="https://github.com/user-attachments/assets/86f957f3-083b-4e65-91cd-3f02a4ee0349" />

    <img width="1919" height="1031" alt="Image" src="https://github.com/user-attachments/assets/f3ab0dfb-bfbe-4873-a087-835d7ef15552" />

    <img width="1914" height="1041" alt="Image" src="https://github.com/user-attachments/assets/a63585e5-c97e-4163-a468-5868dd5f59a2" />

    <img width="1911" height="1036" alt="Image" src="https://github.com/user-attachments/assets/dd28897b-9529-4697-8d87-7a148550c43a" />

## Issues/Clarification
1. Could you clarify if this is the expected visual for the chart? Struggling to understand if this is suppose to be a bar chart. As the y value for the chart is the same. 

    <img width="905" height="475" alt="Image" src="https://github.com/user-attachments/assets/1d35b678-1bb4-48b4-a2ff-1f27a047b43f" />

2. The navigation links at the top of the page contains 3 links(Virtual vs. In-Person, Event Value Estimates, Pariticpaation Trends). At the bottom of the page, there's the three same links. However, there is an extra link being Event Performance Metrics. Is it expected that the "Event Performance Metrics" isn't also in the navigation link at the top?

    <img width="1790" height="325" alt="Image" src="https://github.com/user-attachments/assets/f6ea7a3b-2b38-4f8c-bd80-8d86f206e21c" />

    <img width="1771" height="288" alt="Image" src="https://github.com/user-attachments/assets/ff1a5d24-a7b2-4fca-8d0f-ec11c7733654" />

3. Not sure if this is in the scope of the PR but the dimensions of several UI components change when switching to dark mode. Refer to the attached screenshots below but the summary cards and analytics panels and more are larger in dark mode than in light mode. Theme changes should ideally only affect styling (e.g., colors), not the component sizing or layout.

    <img width="1914" height="1033" alt="Image" src="https://github.com/user-attachments/assets/c62bf0d9-6656-47d4-98f3-23400ccce27b" />

    <img width="1917" height="1036" alt="Image" src="https://github.com/user-attachments/assets/52708bb4-c441-4ebc-af8e-c53af15c08d1" />

DeepighaJ reviewed (changes requested):
- Tested PR locally.
- The bar chart has no tool tip displayed and looks same for all values.
- The subpages need color contrast improvement for texts in dark mode. 
- Found the layout changing issue in dark and white mode as mentioned in previous review comment.
- The Event Performance details Attendence, rating and engagement are compressed into single cell and not shown under respective table header. All texts are white in color whereas light mode has start rating in yellow and value in colors.

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/a025ba71-bf33-46fe-89f4-1cdd6fa8bbc0" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/8b06012c-f69b-4b64-8f35-19fdc982c900" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/2df3677a-2b8a-4d05-953d-69b7416b0864" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/4bc9689f-72b2-4591-88a0-5734b8f830b2" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/57dc7d0d-fd80-4cd3-9eb9-a14bfd240b11" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/3369f020-272b-4e86-b1f6-8c2f9a40981e" />

**The Event Performance details Attendence, rating and engagement are compressed into single cell and not shown under respective table header. All texts are white in color whereas light mode has start rating in yellow and value in colors.**

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/50c07d5b-88ea-468a-b5cd-89437576eceb" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/e9bea85c-0cc8-4803-a809-658214609f70" />

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/534e75f4-b45a-4af2-883e-d89800f2a1e8" />

rithika-paii committed:
fix: bar chart tooltip/scaling, table layout, dark mode contrast

rithika-paii committed:
chore: remove stray files from lint-staged stash
